### PR TITLE
Add fix-your-indent extension

### PR DESCRIPTION
**Pull request by noidwasavailable**
## Description

Adds a new extension `fix-your-indent` which removes unnecessary leading whitespaces when you copy-paste some deeply indented code.

## Screencast

https://github.com/raycast/extensions/assets/26755327/8745be23-c28a-4d6d-83b6-0fc4ec189599

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by noidwasavailable** [comment]
My `package.json` has this issue:

```
Incorrect type. Expected "array".
---
Extension's tools

List of all tools exposed by this extensions.
```

image

Is this why the Extensions / Check (pull_request) might be failing?

File: extensions/fix-your-indent/package.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "eslint": "^8.51.0",
+    "prettier": "^3.0.3",
+    "typescript": "^5.2.2"
+  },
+  "scripts": {
+    "build": "ray build -e dist",
+    "dev": "ray develop",
+    "fix-lint": "ray lint --fix",
+    "lint": "ray lint",
+    "prepublishOnly": "echo \"\\n\\nIt seems like you are trying to publish the Raycast extension to npm.\\n\\nIf you did intend to publish it to npm, remove the \\`prepublishOnly\\` script and rerun \\`npm publish\\` again.\\nIf you wanted to publish it to the Raycast Store instead, use \\`npm run publish\\` instead.\\n\\n\" && exit 1",
```
Comment (pernielsentikaer) [inline extensions/fix-your-indent/package.json:38]:
```suggestion
```
I don't think this is needed

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi 👋 

Thanks for your contribution 💪 

I have now tested your extension, and I have some feedback ready for you:

- metadata images are only meant for showcasing the best views in the extension, since this is a `no-view` command I suggest you move the images to a folder called `media` and include it into the `README` file

I'm looking forward to testing this extension again 🔥 

Request a new review when you are ready. Feel free to contact me here or at [Slack](https://www.raycast.com/community) if you have any questions.

File: extensions/fix-your-indent/package.json (hunk 1 above)
Comment (noidwasavailable) [reply extensions/fix-your-indent/package.json:38]: This was just there by default when I ran the `Create Extension` command. I'll delete it!

**Review by noidwasavailable — COMMENTED** [review]
@pernielsentikaer 

1. Moved files to `media/` and referenced them in `README.md` by ``. It was working well on my markdown previews, hopefully I've done the linking correctly. 
2. Removed `prePublishOnly` command from `package.json`.

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Do you think we should add a preference to let the user set default action (copy/paste) and then add an argument to let the user decide (default/copy/paste)

**Comment by noidwasavailable** [comment]
@pernielsentikaer 

> Do you think we should add a preference to let the user set default action (copy/paste) and then add an argument to let the user decide (default/copy/paste)

I'm interpreting your comment as either the result should get copied to the clipboard or get pasted to the frontmost app. Am I interpreting it correctly?

I'll make some things up when I get some spare time!

**Comment by pernielsentikaer** [comment]
I'm leaving that to you. I can also improve it, and it doesn't have to be in the initial version; that's up to you 🙂

**Comment by mil3na** [comment]
Hi! Is this ready for another review?

**Comment by noidwasavailable** [comment]
@mil3na Hi! Thanks for asking. I'm afraid I don't have much free time nowadays, so I'll consider adding the preferences options in the future. So, **yes, it is ready for a review**.

**Comment by mil3na** [comment]
Hey! Thanks again for your contribution!

I wonder about the need of this extension. Most modern editors have both multiple line edit and auto format. Do you see you needing this kind of extension often? What is the usecase?
Also, have you considered the case where the indentation is not aligned? Just removing lines based on the last line would not be enough.

**Comment by mil3na** [comment]
Have you considered writing a Script Command for it instead? The usecase seems to be too small and too niche for this extension to exist.

**Comment by noidwasavailable** [comment]
@mil3na I'll consider that. I think you can close this for now!

**Pull request closed without merging**